src/outputs.ts

```diff
@@ -0,0 +1,84 @@
+import * as fs from 'fs'
+import * as querystring from 'querystring';
+import { Writable } from 'stream';
+import { highlight } from 'highlight.js';
+
+export class OutputMultiplexer {
+  private _outputs: IOutput[]
+  constructor() {
+    this._outputs = []
+  }
+  add(output: IOutput) {
```

Comment:
nit: pass array in the constructor - adding one half-way does not make much sense.